[TIMOB-25879] Package iOS module zip in "dist" directory (Parity)

hansemannn opened the pull request:
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-25879

hansemannn committed:
[TIMOB-25879] Package iOS module zip in "dist" dir

build commented:
<!--
  1 failure:  Tests have failed...
  0 warning: 
  1 messages
  1 markdown notices
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Tests have failed, see below for more information.
  </td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-9941/1/artifact/dist/mobilesdk-7.2.0.v20180318105617-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  </tbody>
</table>

### Tests: 

| Classname | Name | Time | Error |
| --- | --- | --- | --- |
| ios.Titanium.UI.WebView | loading | 10.004 | file:///Users/build/Librar |

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

sgtcoolguy reviewed (approved):
CR looks fine. But this is a behavioral change and may affect Studio, cc @kolipakakondal @ppedduri 

Also, do we have any actual docs that may need updating if we change the output directory?

hansemannn commented:
We need to update the [iOS Module Project](https://docs.appcelerator.com/platform/latest/#!/guide/iOS_Module_Project-section-src-43288810_iOSModuleProject-Buildandpackagethemodule) guide that talks about the output directory. It will be updated once this one is merged to keep in sync. Not sure about the Studio impact, but I don't see issues that this change could cause there.

kolipakakondal commented:
Added studio necessary changes with this https://github.com/appcelerator/titanium_studio/pull/1067 

Based on the SDK 7.3.0 release plan - studio can include this change

hansemannn committed:
Ensure "dist" directory exists before packaging

Prevents the following error:

ENOENT: no such file or directory, open '/Users/hknoechel/Documents/appcelerator_modules/ti.nfc/ios/dist/ti.nfc-iphone-2.0.1.zip'
ENOENT: no such file or directory, open '/Users/hknoechel/Documents/appcelerator_modules/ti.nfc/ios/dist/ti.nfc-iphone-2.0.1.zip'

sgtcoolguy committed:
Merge branch 'master' into TIMOB-25879

build commented:
<!--
  0 failure: 
  0 warning: 
  1 messages
  
  DangerID: danger-id-default;
-->



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-9941/3/artifact/dist/mobilesdk-7.4.0.v20180619062327-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

hansemannn committed:
Merge branch 'master' into TIMOB-25879

hansemannn committed:
Merge branch 'master' into TIMOB-25879

build commented:
<!--
  0 failure: 
  0 warning: 
  1 messages
  
  DangerID: danger-id-default;
-->



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-9941/6/artifact/dist/mobilesdk-7.4.0.v20180621081310-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>
